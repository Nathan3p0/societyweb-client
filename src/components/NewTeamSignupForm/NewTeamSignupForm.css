.signup__form li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: .5rem
}

.signup__form > li > label {
    flex: 1 0 100px;
}

.signup__form > li > label + * {
    flex: 1 0 220px;
}

.signup__form li button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: #333333;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.signup__form li button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}

.signup__form li input {
    padding: 15px;
    border: none;
}

.formError {
    color: #D62246;
    max-width: 320px;
    width: 100%;
    text-align: right;
    margin-left: auto;
}