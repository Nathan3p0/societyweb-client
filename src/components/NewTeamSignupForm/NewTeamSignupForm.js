import React from 'react';
import './NewTeamSignupForm.css';

const NewTeamSignupForm = (props) => {
    return (
        <form onSubmit={props.handleSubmit}>
            <ul className="signup__form">
                <li>
                    <label htmlFor="fullname">Full Name:</label>
                    <input type="text" name="full_name" id="fullname" placeholder="name" required />
                </li>
                <li>
                    <label htmlFor="username">Username:</label>
                    <input type="text" name="username" id="username" placeholder="username" required />
                </li>
                <li>
                    <label htmlFor="password">Password:</label>
                    <input type="password" name="password" id="password" placeholder="password" required />
                </li>
                <li>
                    <label htmlFor="email">Email:</label>
                    <input type="text" name="email" id="email" placeholder="email" required />
                </li>
                <li>
                    <label htmlFor="phone">Phone Number:</label>
                    <input type="tel" name="phone" id="phone" placeholder="123-455-0505" required />
                </li>
                <li>
                    <label htmlFor="group_name">Team Name</label>
                    <input type="text" name="group_name" id="group_name" placeholder="team name" required />
                </li>
                <li>
                    <button type="submit">Sign Up</button>
                </li>
            </ul>
        </form>
    );
}

export default NewTeamSignupForm;