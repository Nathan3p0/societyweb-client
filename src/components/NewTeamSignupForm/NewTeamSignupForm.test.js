import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NewTeamSignupForm from './NewTeamSignupForm'

describe('Member Login Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NewTeamSignupForm />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<NewTeamSignupForm />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})