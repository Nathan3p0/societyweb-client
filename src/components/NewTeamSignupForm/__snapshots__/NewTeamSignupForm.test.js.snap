// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Team Signup Component Rendering Tests renders to UI as expected 1`] = `
<form>
  <ul
    className="signup__form"
  >
    <li>
      <label
        htmlFor="fullname"
      >
        Full Name:
      </label>
      <input
        id="fullname"
        name="full_name"
        placeholder="name"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        id="username"
        name="username"
        placeholder="username"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
    </li>
    <li>
      <label
        htmlFor="email"
      >
        Email:
      </label>
      <input
        id="email"
        name="email"
        placeholder="email"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="phone"
      >
        Phone Number:
      </label>
      <input
        id="phone"
        name="phone"
        placeholder="123-455-0505"
        required={true}
        type="tel"
      />
    </li>
    <li>
      <label
        htmlFor="group_name"
      >
        Team Name
      </label>
      <input
        id="group_name"
        name="group_name"
        placeholder="team name"
        required={true}
        type="text"
      />
    </li>
    <li>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </li>
  </ul>
</form>
`;
